File: springboot-mall/src/main/java/com/tonyydl/springbootmall/data/dto/ProductDTO.java
package com.tonyydl.springbootmall.data.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.tonyydl.springbootmall.constant.ProductCategory;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(toBuilder = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonPropertyOrder({
        "product_id",
        "product_name",
        "category",
        "image_url",
        "price",
        "stock",
        "description",
        "created_date",
        "last_modified_date"
})
public class ProductDTO {

    @JsonProperty("product_id")
    private Integer productId;

    @JsonProperty("product_name")
    private String productName;

    private ProductCategory category;

    @JsonProperty("image_url")
    private String imageUrl;

    private Integer price;

    private Integer stock;

    private String description;

    @JsonProperty("created_date")
    private long createdDate;

    @JsonProperty("last_modified_date")
    private long lastModifiedDate;
}

File: android-app/app/src/main/java/com/tonyydl/shoppingmallapp/data/dto/ProductDTO.kt
package com.tonyydl.shoppingmallapp.data.dto

import com.google.gson.annotations.SerializedName
import com.tonyydl.shoppingmallapp.data.Mapper
import com.tonyydl.shoppingmallapp.data.ProductCategory
import com.tonyydl.shoppingmallapp.data.vo.Product

data class ProductDTO(
    @SerializedName("product_id") val productId: Int = 0,
    @SerializedName("product_name") val productName: String? = null,
    val category: ProductCategory? = null,
    @SerializedName("image_url") val imageUrl: String? = null,
    val price: Int = 0,
    val stock: Int = 0,
    val description: String? = null,
    @SerializedName("created_date")
    val createdDate: Long? = null,
    @SerializedName("last_modified_date")
    val lastModifiedDate: Long? = null
) : Mapper<Product> {
    override fun transform(): Product {
        return Product(
            productId = productId,
            productName = productName.orEmpty(),
            category = category ?: ProductCategory.NONE,
            imageUrl = imageUrl.orEmpty(),
            price = price,
            stock = stock,
            description = description.orEmpty(),
            createdDate = createdDate ?: 0L,
            lastModifiedDate = lastModifiedDate ?: 0L
        )
    }
}